src/proxy/reconnect.rs

```diff
@@ -44,9 +47,34 @@ where
 }
 
 #[derive(Clone, Debug)]
-enum Backoff {
+pub enum Backoff {
     None,
-    Fixed(Duration),
+    Exponential {
+        min: Duration,
+        max: Duration,
+        jitter: f64,
+    },
+}
+
+impl Backoff {
+    fn for_failures<R: rand::Rng>(&self, failures: u32, mut rng: R) -> Option<Duration> {
+        match self {
+            Backoff::None => None,
+            Backoff::Exponential { max, min, jitter } => {
+                let backoff = Duration::min(min.mul(2_u32.pow(failures)), *max);
+
+                if *jitter != 0.0 {
+                    Some(backoff)
+                } else {
+                    let rand_jitter = rng.gen::<f64>() * jitter;
```

Comment:
A random f64 could be negative, right? I *think* that'd be a problem, probably when casting the millis back to a u64.

Reply:
Maybe a `quickcheck!` for `Backoff::for_failures` would trigger something bad?

Reply:
While I also expected that to be the case, I found out the [docs](https://github.com/rust-random/rand/blob/master/src/lib.rs#L32) pointing out that this actually should give you a random float between 0.0 and 1.0. Will run it through quickcheck though...

Reply:
Oh my, I should have looked, but that's not obvious at all. I suppose adding a `debug_assert` would help document this behavior for us.